README.md

```diff
@@ -1,31 +1,118 @@
+
+The codebase supports the following workflow to sample a catalyst+adsorbate system. We start by selecting the number of elements in the bulk material. A bulk structure with this number of elements is selected at random.  A surface is then randomly selected from all symmetrically distinct surfaces from the material with Miller indices less than or equal to 2. A randomly selected adsorbate is then placed over this surface.
+
+In addition to sampling a single catalyst+adsorbate system, we could also enumerate multiple different inputs. At each step, “select” may mean randomly sample or choose a specified object as input by the user, as described in further detail in the next section.
+
+![Workflow image](workflow_image.png)
+
+## APIs supported
 
 These are the supported use cases and how to run them:
 
-1. Generate VASP input files for one random adsorbate/bulk/surface/config combination, based on a specified random seed. For the following example, files will be in two directories, `outputs/random0_surface/` and `outputs/random0_adslab/`.
+1. Generate VASP input files for one random adsorbate/bulk/surface/config combination, based on a specified random seed. This follows the workflow described in the diagram above, where every step to “select” a material is randomly sampled. The provided bulk database should be a dict of {number of elements: list of corresponding bulks}. For the following example, files will be in two directories, `outputs/random0/surface/` and `outputs/random0/adslab/`.
+
 ```
-python sample_structure.py --bulk_db bulk_database.pkl --adsorbate_db adsorbate_database.pkl --output_dir outputs/ --seed 0
+python sample_structure.py --bulk_db bulk_db_nelems.pkl \
+  --adsorbate_db adsorbate_db.pkl  \
+  --output_dir outputs/ --seed 0
 ```
 
-2. Generate one specified adsorbate, one more more specified bulks, and all possible surfaces and configs. The following example generates files for all adslab placements of adsorbate 10 on all possible surfaces from bulk 20. Files will be stored in `outputs/10_20_0/surface/`, `outputs/10_20_0/adslab0/`, `outputs/10_20_0/adslab1/`, ..., `outputs/10_20_1/surface/`, `outputs/10_20_1/adslab0/`, and so on for all combinations of surfaces and adslab configs. You may also choose multiple bulks (and all of their surfaces and adslab configs) by giving a comma separated list of bulk indices.
+1. Generate one specified adsorbate, one more more specified bulks, and all possible surfaces and configs. The following example generates files for all adsorbate placements of adsorbate 10 on all possible surfaces from bulk 20. Files will be stored in `outputs/10_20_0/surface/`, `outputs/10_20_0/adslab0/`, `outputs/10_20_0/adslab1/`, ..., `outputs/10_20_1/surface/`, `outputs/10_20_1/adslab0/`, and so on for all combinations of surfaces and adsorbate+surface configs. You may also choose multiple bulks (and all of their surfaces and adsorbate+surface configs) by giving a comma separated list of bulk indices.
```

Comment:
Update numberings - 1,2,3 instead of all 1

Reply:
for enumerating specific adsorbate on a specific bulk (instead of a random one). I think it'd be really helpful to have a mapping of the indices to mpids (and indices to smile string), or some sort of helper script/instruction on how to get the index based on mpid/smile string. 

If someone did want to enumerate for a specific adsorbate bulk, I assume they are going to have an adsorbate and mpid in mind. (Based on experience, it was not straightforward for me to figure out what index to use, especially for bulk).